MAINTAINERS.md: add release checklist (#6274)
urbit/urbit: MAINTAINERS.md

- [ ] Merge `~zod`'s `%base` desk into its `%kids` desk to distribute the new code to the network: `|merge %kids our %base, =gem %only-that`
- [ ] Before exiting the screen session on `~zod`, make sure the screen session is not left in copy mode for a long period of time, since that will disrupt `~zod`'s operation.

### Release Communications

- [ ] Tag the commit that went onto the live network as a release, using GitHub's "Releases" interface.  See the "Tagging" section of this document for details.
- [ ] Email the urbit-dev mailing list with an "[ANNOUNCE]" prefix in the email subject.  See previous release emails for reference.
- [ ] Tweet from the `@zodisok` Twitter account linking to the GitHub release.
- [ ] Post links to the release in the Urbit Community Development channel and a channel in the UF public group.

### Post-Release Git Cleanup

> - [ ] Merge master back into develop.

Reviewer: ```suggestion
- [ ] Merge `master` back into `develop`.
```

> - [ ] Cut a new release branch from develop.  TODO describe release branch naming

Reviewer: ```suggestion
- [ ] Cut a new release branch from `develop`.  The branch should have the format `release/urbit-os-vX.XX`
```
